interlok-apache-http5/src/main/java/com/adaptris/core/http/apache5/DefaultClientBuilder.java

```diff
@@ -92,13 +92,15 @@ protected HttpClientBuilder customiseTimeouts(HttpClientBuilder builder, long ti
     RequestConfig.Builder requestCfg = RequestConfig.custom();
     if (getConnectTimeout() != null) {
       requestCfg.setConnectTimeout(Timeout.ofMilliseconds(getConnectTimeout().toMilliseconds()));
+      requestCfg.setConnectionRequestTimeout(Timeout.ofMilliseconds(getConnectTimeout().toMilliseconds()));
+    } else if (timeout >= 0) {
+      requestCfg.setConnectTimeout(Timeout.ofMilliseconds(timeout));
+      requestCfg.setConnectionRequestTimeout(Timeout.ofMilliseconds(timeout));
     }
     if (getReadTimeout() != null) {
       requestCfg.setResponseTimeout(Timeout.ofMilliseconds(getReadTimeout().toMilliseconds()));
-    }
-    if (timeout != HttpProducer.DEFAULT_TIMEOUT) {
+    } else if (timeout >= 0) {
       requestCfg.setResponseTimeout(Timeout.ofMilliseconds(timeout));
-      //builder.setDefaultRequestConfig(SocketConfig.custom().setSoTimeout(Timeout.ofMilliseconds(timeout)).build());
```

Comment:
@albinoloverats I actually don't know, but I just compared with your changes on apache-http and why here we don't do:

`builder.setDefaultRequestConfig(SocketConfig.custom().setSoTimeout(Timeout.ofMilliseconds(timeout)).build());`

and same above with getReadTimeout?

Reply:
Do what, set the `SocketConfig`? This could be an oversight on my part; I'll take a look.

Reply:
Yeah like in https://github.com/adaptris/interlok-apache-http/pull/163/files#diff-9503b2c9668b81e26543a78b89966534356c35d4d0bf82b643794fc7232fb889R100

In  interlok-apache-http/src/main/java/com/adaptris/core/http/apache/DefaultClientBuilder.java line 100 and 103

Reply:
It looks like the Apache5 `HttpClientBuilder` does not have a `setDefaultSocketConfig` method, so it's a bit more convoluted now (creating a connection manager).

Reply:
Ha fair enough then